test/Validate.SSA.cpp

```diff
@@ -1014,5 +1014,148 @@ TEST_F(ValidateSSA, PhiMissingLabelBad) {
   EXPECT_THAT(getDiagnosticString(), HasSubstr("missing"));
 }
 
-// TODO(umar): OpGroupMemberDecorate
+TEST_F(ValidateSSA, IdDominatesItsUseGood) {
```

Comment:
This is the case where uses are in different blocks from their definitions.
I'd also add another case where a use is in the same block as its definition.  (Or add that case to this one.)